elasticsearch_dsl/search.py

```diff
@@ -294,7 +295,10 @@ def _clone(self):
 
         s._response_class = self._response_class
         s._sort = self._sort[:]
-        s._source = self._source.copy() if self._source else None
+        if hasattr(self._source, "copy"):
```

Comment:
I left it if something needs special care. but I don't know any special case atm.